src/linux/linux-host-session.vala

```diff
@@ -1219,6 +1234,13 @@ namespace Frida {
 			if (entrypoint == null)
 				entrypoint = new DefaultActivityEntrypoint ();
 
+			if (aux_options.contains ("userid")) {
+				string? string_userid = null;
+				if (!aux_options.lookup ("userid", "s", out string_userid))

```

Comment:
Here we should be expecting a uint, not a string.

Reply:
Honestly, I tried several ways but I could not manage to address this bit. 
Using any numeric format string always fails, also when manually setting the type via python

Reply:
Out of all of the types supported by [GVariant](https://developer.gnome.org/glib/stable/gvariant-format-strings.html), I elected to only support three types: `string`, `int64`, and `boolean`. We could expand on this in the future, as long as we try to keep it to a minimum in order to ease the burden on each language binding.

Something like this should work:

```
				int64 uid = 0;
				if (!aux_options.lookup ("uid", "x", out uid))
```

(And then cast that to `uint` afterwards.)

Reply:
Oh ok, thanks very much for the explanation. 
Addressed by abf41cf2b57c1af01cb400b4a6ff3a986e3bfb8d